File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/RegexRule.java
package com.wesleyelliott.kubwa.rule;

import android.text.TextUtils;

/**
 * Created by wesley on 2016/07/28.
 */

public class RegexRule extends Rule {

    private String regex;

    public RegexRule(String regex) {
        this.regex = regex;
    }

    protected String getRegex() {
        return regex;
    }

    @Override
    public boolean isValid(String value) {
        return !TextUtils.isEmpty(value) && value.matches(getRegex());
    }
}

File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/RegexRule.kt
package com.wesleyelliott.kubwa.rule

import com.wesleyelliott.kubwa.Utils

/**
 * Created by wesley on 2016/07/28.
 */

open class RegexRule(protected val regex: String) : Rule<String>(String::class.java) {

    override fun isValid(value: String?): Boolean {
        return value != null && !Utils.isEmpty(value) && value.matches(regex.toRegex())
    }
}
